akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/PersistentBehaviors.scala

```diff
@@ -137,19 +87,62 @@ object PersistentBehaviors {
   }
 }
 
-class PersistentBehavior[Command, Event, State](
-  @InternalApi private[akka] val persistenceIdFromActorName: String ⇒ String,
-  val initialState:                                          State,
-  val commandHandler:                                        PersistentBehaviors.CommandHandler[Command, Event, State],
-  val eventHandler:                                          (State, Event) ⇒ State,
-  val recoveryCompleted:                                     (ActorContext[Command], State) ⇒ Unit,
-  val tagger:                                                Event ⇒ Set[String],
-  val snapshotOn:                                            (State, Event, Long) ⇒ Boolean) extends UntypedBehavior[Command] {
-
-  import PersistentBehaviors._
-
-  /** INTERNAL API */
-  @InternalApi private[akka] override def untypedProps: akka.actor.Props = PersistentActorImpl.props(() ⇒ this)
+@DoNotInherit
+class PersistentBehavior[Command, Event, State] private (
```

Comment:
Can we hide the implementation of this class in the implementation? Seems like a potential compatibility problem that we expose all the fields in public API. If we want to support the builder style with the deferred behavior we should do so by exposing here an interface that allows access to exactly the methods we'd like. E.g. like

```scala
trait PersistentBehavior[Command, Event, State] extends Behavior[Command] {
  def snapShotWhen = ...
   ...
}
```

Or do we want to support the `new PersistentBehavior` style for some reason (or why is the `this` constructor public at all)?

Reply:
I had a quick try here: https://github.com/ktoso/akka/pull/34/commits/bffa7287f5b8a42f9bc6542e03cffa33a9d5738b

Reply:
Yeah sounds fine